services/ui-src/src/components/reports/ModalOverlayReportPage.tsx

```diff
@@ -65,9 +73,24 @@ export const ModalOverlayReportPage = ({
 
   // Display Variables
   let reportFieldDataEntities = report?.fieldData[entityType] || [];
+
   (reportFieldDataEntities as any[]).map(
     (entity) => (entity["isOtherEntity"] = true)
   );
+
+  reportFieldDataEntities = reportFieldDataEntities.filter(
+    (entity: EntityShape) => {
+      if (
+        Object.keys(entity).findIndex((key: string) =>
+          key.includes(entityType)
+        ) > 0
+      ) {
+        return entity;
+      }
+      return;
+    }
+  );
+
```

Comment:
Whats happening here? Can we move this to a function? And should the filter be returning false instead of just return at the very end?